[Eris Jacey] iP

erisjacey opened the pull request.

Commit:
Add Gradle support

erisjacey committed:
Add Greet, Echo, Exit

erisjacey committed:
Fix small bug regarding Scanner input

erisjacey committed:
Add Add, List

erisjacey committed:
Add Mark as Done

erisjacey committed:
Add ToDos, Events, Deadlines

erisjacey committed:
Add Automated Text UI Testing

erisjacey committed:
Add Handle Errors

erisjacey committed:
Add various comments

erisjacey committed:
Duke.java: modify addTask

erisjacey committed:
Add Delete

erisjacey committed:
Add Enums (TaskType)

erisjacey committed:
Replace "?" with respective "✓" and "✘"

erisjacey committed:
Change chatbot logo and name

erisjacey committed:
runtest.bat: Clean up script

erisjacey committed:
ToDo.java: Clean up toString method

erisjacey committed:
Modify Exception implementation

erisjacey committed:
Modify Exception implementation

erisjacey committed:
Duke.java: Abstract away logo

erisjacey committed:
Add Level-7: Save

erisjacey committed:
Level-8: Add Dates and Times

erisjacey committed:
Merge branch 'branch-Level-7'

erisjacey committed:
Merge branch 'branch-Level-8'

erisjacey committed:
A-MoreOOP: Replace Printer with Ui

erisjacey committed:
A-MoreOOP: Add TaskList

erisjacey committed:
A-MoreOOP: Add Parser and Command

erisjacey committed:
A-MoreOOP: Modify Duke.java

erisjacey committed:
A-Packages: Organize into Packages

erisjacey committed:
A-JUnit: Add JUnit Tests (TaskListTest.java)

erisjacey committed:
A-Jar: Create a JAR File

erisjacey committed:
A-JavaDoc: Add JavaDoc

erisjacey committed:
A-CodingStandard: Follow the Coding Standard

erisjacey committed:
Level-9: Add Find

erisjacey committed:
Merge branch 'branch-A-JavaDoc'

erisjacey committed:
Merge branch 'branch-A-CodingStandard'

erisjacey committed:
Merge branch 'branch-Level-9'

erisjacey committed:
Storage.java: Fix bug regarding the creation of save file directories

erisjacey committed:
Fix code style for switch-case statements

erisjacey committed:
Add Clear

erisjacey committed:
Add Help and CommandType

erisjacey committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

erisjacey committed:
CommandType.java: fix a small bug regarding DEADLINE.toString()

erisjacey committed:
A-Gradle: Use Gradle

erisjacey committed:
Merge branch 'branch-A-Gradle'

erisjacey committed:
A-CheckStyle: Use CheckStyle to fix coding style violations

erisjacey committed:
Merge branch 'branch-A-CheckStyle'

erisjacey committed:
Iteration 1: add DialogBox

erisjacey committed:
Iteration 2: add custom behavior to DialogBox

erisjacey committed:
Add most functionalities to the GUI (except "bye")

erisjacey committed:
Add padding

* between each DialogBox
* between each ImageView and its Label

erisjacey committed:
Refractor code to include CommandResult

erisjacey committed:
Level-10: integrate GUI using JavaFX

erisjacey committed:
Merge branch 'branch-Level-10'

erisjacey committed:
Refractor code into more packages

* messages: Output (rename from Ui)
* parsers: Parser
* storage: Storage
* task: TaskList

erisjacey committed:
Move JUnit tests to correct package

soaza commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,37 @@
+            throws DukeInvalidTaskTimeException,
+            DukeInvalidTaskDescriptionException,
+            DukeLoadingErrorException {
+        Task toAdd = tasks.addTask(tag, input);
```
Maybe leave a line here for better clarity?

erisjacey replied:
What do you mean? Do you mean leave a line between line 30 and 31?

soaza commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public CommandResult execute(TaskList tasks, Output output, Storage storage)
+            throws DukeInvalidListNumberInputException, DukeLoadingErrorException {
```
Leave a line here for better clarity.

erisjacey replied:
Where exactly do I leave a line?

soaza reviewed (commented):
Overall well implemented Eris! I like the use of abstract classes and enums. Also, I liked how you designed it such that each class is small, compact and readable. This is very convenient for reviewers to read. Good job man! LGTM.

ngmarcus commented on src/main/java/duke/messages/Output.java:
```diff
@@ -0,0 +1,191 @@
+            + "/__/     \\__\\  \\__/  \\__/     |_______|_"
+            + "______/     \\______/  |__|  |__| |_______|        \\______/\n";
+    private static final String indent = "    ";
+    private static final String border = "_________________________________________________________";
```
Should border be capitalised as well?

erisjacey replied:
Good spotting! Thanks!

ngmarcus commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public CommandResult execute(TaskList tasks, Output output, Storage storage)
+            throws DukeInvalidListNumberInputException, DukeLoadingErrorException {
```
Would it be better if the DukeInvalidListNumberInputException was given a slightly shorter name, perhaps DukeInvalidIndexException?

erisjacey replied:
That makes sense!

ngmarcus commented on src/main/java/duke/messages/Output.java:
```diff
@@ -0,0 +1,191 @@
+            + "--)   |   |  `--'  | |  |  |  | |  |____        |  `--'  |\n"
+            + "/__/     \\__\\  \\__/  \\__/     |_______|_"
+            + "______/     \\______/  |__|  |__| |_______|        \\______/\n";
+    private static final String indent = "    ";
```
Should the indent variable be capitalised if it was declared static final?

ngmarcus commented on src/main/java/duke/messages/Output.java:
```diff
@@ -0,0 +1,191 @@
+ */
+public class Output {
+
+    private static final String logo = "     ___   ____    __    ____  _______     _______."
```
Would it be better to name constant (static final) fields with upper case letters?

ngmarcus reviewed (commented):
I like your overall adherence to the Java coding standard!

erisjacey committed:
Abstract messages away in Message.java

erisjacey committed:
Create more JUnit tests (duke/exceptions)

erisjacey committed:
Tweak GUI

Add the following:
- Colours
- Background image
- Chat bubble (rounded)
- New font (Lucida Console)

erisjacey committed:
Change package name "duke/task" to "duke/tasks"

erisjacey committed:
Prepare the generation of a new jar file

erisjacey committed:
A-Assertions: Use Assertions

erisjacey committed:
A-CodeQuality: Improve Code Quality

erisjacey committed:
Add more assert statements

erisjacey committed:
A-Lambdas: Use Lambdas

erisjacey committed:
A-Lambdas: Use Lambdas

erisjacey committed:
Revert "A-Lambdas: Use Lambdas"

This reverts commit 387229ca933e9197a9f8c2792d90000d3d1de6df.

erisjacey committed:
A-Streams: Use Streams

erisjacey committed:
Merge pull request #1 from erisjacey/branch-A-Assertions

A-Assertions: Use Assertions

erisjacey committed:
Merge branch 'master' of https://github.com/erisjacey/ip

erisjacey committed:
Merge branch 'master' into branch-A-CodeQuality

erisjacey committed:
Merge branch 'master' into branch-A-Lambdas

erisjacey committed:
Merge branch 'master' into branch-A-Streams

erisjacey committed:
Merge pull request #2 from erisjacey/branch-A-CodeQuality

A-CodeQuality: Improve Code Quality

erisjacey committed:
Merge branch 'master' of https://github.com/erisjacey/ip into branch-A-CodeQuality

erisjacey committed:
Merge branch 'branch-A-CodeQuality'

erisjacey committed:
Merge branch 'master' into branch-A-Streams

erisjacey committed:
Merge branch 'master' into branch-A-Lambdas

erisjacey committed:
Merge pull request #3 from erisjacey/branch-A-Lambdas

A-Lambdas: Use Lambdas

erisjacey committed:
Merge branch 'master' of https://github.com/erisjacey/ip

erisjacey committed:
Merge branch 'master' into branch-A-Streams

erisjacey committed:
Merge pull request #4 from erisjacey/branch-A-Streams

A-Streams: Use Streams

erisjacey committed:
Merge branch 'master' of https://github.com/erisjacey/ip

erisjacey committed:
A-CI: Set up CI

erisjacey committed:
B-ViewSchedules: Add a way to view schedule

erisjacey committed:
Merge pull request #5 from erisjacey/branch-B-ViewSchedules

B-ViewSchedules: Add a way to view schedule

erisjacey committed:
Merge branch 'master' of https://github.com/erisjacey/ip

erisjacey committed:
gradle.yml: Remove Ubuntu and Mac for CI

erisjacey committed:
Remove all wildcart imports

- duke.command.Command
- duke.command.ScheduleCommand
- parsers.Parser

erisjacey committed:
TaskListTest.java: fix checkstyle errors

erisjacey committed:
style.css: rewrite author reuse

erisjacey committed:
Message.java: edit default message(s)

- MESSAGE_SCHEDULE_INTRO

erisjacey committed:
Upload representative screenshot

erisjacey committed:
Set theme jekyll-theme-architect

erisjacey committed:
Update README.md

Rewrote user guide to completed updated version

erisjacey committed:
Merge remote-tracking branch 'origin/master'

erisjacey committed:
Message.java: further customize chatbot

erisjacey committed:
Merge branch 'master' of https://github.com/erisjacey/ip

erisjacey committed:
Update README.md

erisjacey committed:
Update README.md

erisjacey committed:
Add section links

erisjacey committed:
User guide: add expected outcome pictures

erisjacey committed:
README.md: add pictures

erisjacey committed:
Update README.md
